src/phases/query/CollScanSimplifiablePredicate.yml

```diff
@@ -0,0 +1,360 @@
+                {^Array: {of:
+                  {b: {$gt: {^Inc: {start: 10}}}},
+                  number: {^Parameter: {Name: "Width", Default: -1}}
+                }}
+            }}
+          }}}
+
+- TemplateName: SimplifiableElemMatchCNF
+  Config:
+    ActorFromTemplate:
+      TemplateName: FindQueryTemplate
+      TemplateParameters:
+        Name: {^Parameter: {Name: "Name", Default: "unused"}}
+        ActivePhase: {^Parameter: {Name: "ActivePhase", Default: -1}}
+        Filter:
+          {arr_of_objs: {$elemMatch: {$and:
+            {^Cycle: {
+              ofLength: 1,
+              fromGenerator:
+                {^Array: {of:
+                  {$or: [
+                    {b: {$eq: -1}},
+                    {c: {$eq: 0}},
+                    {nested: {f1: {$gte: {^Inc: {start: 10}}}}},
+                    {nested: {f2: {$lte: {^Inc: {start: 0, step: -1}}}}},
+                    {nested: {f3: {$lt: {^Inc: {start: 0, step: -1}}}}},
+                    {nested: {f4: {$gt: {^Inc: {start: 10}}}}},
+                  ]},
+                  number: {^Parameter: {Name: "Width", Default: -1}}
+                }}
+            }}
+          }}}
+
+- TemplateName: SimplifiableElemMatchDNF
+  Config:
+    ActorFromTemplate:
+      TemplateName: FindQueryTemplate
+      TemplateParameters:
+        Name: {^Parameter: {Name: "Name", Default: "unused"}}
+        ActivePhase: {^Parameter: {Name: "ActivePhase", Default: -1}}
+        Filter:
+          {arr_of_objs: {$elemMatch: {$or:
+            {^Cycle: {
+              ofLength: 1,
+              fromGenerator:
+                {^Array: {of:
+                  {$and: [
+                    {b: {$eq: -1}},
+                    {c: {$eq: 0}},
+                    {nested: {f1: {$gte: {^Inc: {start: 10}}}}},
+                    {nested: {f2: {$lte: {^Inc: {start: 0, step: -1}}}}},
+                    {nested: {f3: {$lt: {^Inc: {start: 0, step: -1}}}}},
+                    {nested: {f4: {$gt: {^Inc: {start: 10}}}}},
+                  ]},
+                  number: {^Parameter: {Name: "Width", Default: -1}}
+                }}
+            }}
+          }}}
+
+Actors:
+- Name: ClearCollection
+  Type: CrudActor
+  Database: *Database
+  Threads: 1
+  Phases:
+    OnlyActiveInPhases:
+      Active: [0]
+      NopInPhasesUpTo: *MaxPhases
+      PhaseConfig:
+        Repeat: 1
+        Threads: 1
+        Collection: *Collection
+        Operations:
+        - OperationName: drop
+
+- Name: InsertData
+  Type: Loader
+  Threads: 4
+  Phases:
+    OnlyActiveInPhases:
+      Active: [1]
+      NopInPhasesUpTo: *MaxPhases
+      PhaseConfig:
+        Repeat: 1
+        Database: *Database
+        MultipleThreadsPerCollection: true
+        CollectionCount: 1
+        DocumentCount: *DocumentCount
+        BatchSize: 1000
+        Document:
+          # None of these match any of the filters in this workload, forcing each query to scan 
+          # the entire collection.
```

Comment:
I think the query will scan the entire collection regardless since we have no indexes (we have exactly one possible access path plan)? Maybe remove the last part of the comment.

As a note, this will be interesting when it comes to testing histogram CE (M4 or even later, not in M2) since it would potentially be able to tell us that the CE for all these queries is 0 more accurately.

Reply:
Good point, I removed that part.